packages/models.py

```diff
@@ -0,0 +1,155 @@
+        return f"{self.city}, {self.street_name} {self.street_number}"
+
+
+class Partner(models.Model):
+    name = models.CharField(max_length=64, unique=True, primary_key=True)
+    rates = models.CharField(max_length=32, default='0')
+    address = models.ForeignKey(Address, on_delete=models.CASCADE)
+
+    def get_rates(self):
+        return [int(rate) for rate in str(self.rates).split(',')]
+
+    def related_orders(self):
+        return Order.objects.filter(partner=self)
+
+    def __str__(self):
+        return self.name
+
+    def __eq__(self, other):
+        return str(self) == str(other)
+
+
+class Order(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    collection_date = models.DateTimeField(auto_now_add=True)
+    partner = models.ForeignKey(Partner, on_delete=models.CASCADE)
+    notes = models.TextField(blank=True)
+
+    def related_packages(self):
+        return Package.objects.filter(order=self)
+
+    def overall_package_status(self):
+        packages = list(self.related_packages())
+        packages.sort()
+
+        if packages:
+            return packages[0].status_obj()
+        else:
+            return Package.Status.WAIT
+
+    def status_color(self):
+        return Package.Status(self.overall_package_status()).color()
+
+    def __str__(self):
+        return f"{self.partner} {self.collection_date}"
+
+    def __lt__(self, other):
+        return self.collection_date > other.collection_date
+
+
+class Package(models.Model):
+    class Status(enum.Enum):
+        WAIT = 'Awaiting collection'
+        STORED = 'Warehoused'
+        ON_ROUTE = 'On route to destination'
+        DELIVERED = 'Delivered'
+
+        def next(self):
+            members = list(self.__class__)
+            index = members.index(self) + 1
+            if index >= len(members):
+                index = len(members) - 1
+            return members[index]
+
+        def prev(self):
+            members = list(self.__class__)
+            index = members.index(self) - 1
+            if index < 0:
+                index = 0
+            return members[index]
+
+        def color(self):
+            members = list(StatusColor)
+            index = list(self.__class__).index(self)
+
+            return members[index].value
+
+        @classmethod
+        def choices(cls):
+            return [(i.name, i.value) for i in cls]
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    status = models.CharField(max_length=32, choices=Status.choices(), default='WAIT')
+    origin = models.ForeignKey(Address, on_delete=models.CASCADE, related_name='origin')
+    destination = models.ForeignKey(Address, on_delete=models.CASCADE, related_name='destination')
+    order = models.ForeignKey(Order, on_delete=models.CASCADE)
+    rate = models.DecimalField(default=0, max_digits=5, decimal_places=2)
+    full_name = models.CharField(max_length=32, blank=True)

```

Comment:
32 characters for a full name?
What about [this guy](https://www.google.com/search?q=longest+full+name&rlz=1C1GCEU_enIL928IL928&oq=longest+full+name&aqs=chrome..69i57j0j0i22i30l5j0i390.4167j0j7&sourceid=chrome&ie=UTF-8)

Reply:
Do I really have to care about this dude or any other dude that has such a long name? (he's dead btw)

Reply:
My point was that it seems like a strict limit, It would be a shame if that would cause a bug in the future. Especially since there is no reason not to provide more.

Reply:
https://stackoverflow.com/questions/1261441/what-size-to-pick-for-a-nvarchar-column

Reply:
I replaced it with 64, in this context it's fine (since it's about names) I don't really want people to be able to make such long names. (formatting and stuff)